packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -528,18 +663,153 @@ abstract class SliderComponentShape {
+  static const double _thumbGap = 2.0;
+
+  @override
   void paint(
     PaintingContext context,
-    Offset thumbCenter, {
-    Animation<double> activationAnimation,
+    Offset offset, {
+    RenderBox parentBox,
+    SliderThemeData sliderTheme,
     Animation<double> enableAnimation,
+    List<double> trackDividers,
+    List<bool> trackSegmentsActive,
     bool isDiscrete,
-    TextPainter labelPainter,
-    RenderBox parentBox,
+    bool isEnabled,
+  }) {
+    assert(trackDividers.isNotEmpty);
+
+    final Rect trackRect = getPreferredRect(
+      parentBox: parentBox,
+      offset: offset,
+      sliderTheme: sliderTheme,
+      isEnabled: isEnabled,
+      isDiscrete: isDiscrete
+    );
+
+    // Used to create a gap around the thumb iff the slider is disabled.
+    double horizontalAdjustment = 0.0;
+    if (!isEnabled) {
+      final double thumbRadius = sliderTheme.thumbShape.getPreferredSize(isEnabled, isDiscrete).width / 2.0;
+      final double gap = _thumbGap * (1.0 - enableAnimation.value);
+      horizontalAdjustment = thumbRadius + gap;
+    }
+
+    final ColorTween activeTrackColorTween = ColorTween(begin: sliderTheme.disabledActiveTrackColor , end: sliderTheme.activeTrackColor);
+    final ColorTween inactiveTrackColorTween = ColorTween(begin: sliderTheme.disabledInactiveTrackColor , end: sliderTheme.inactiveTrackColor);
+    final Paint activePaint = Paint()..color = activeTrackColorTween.evaluate(enableAnimation);
+    final Paint inactivePaint = Paint()..color = inactiveTrackColorTween.evaluate(enableAnimation);
+
+    final List<double> allTrackDividers = <double>[]..add(0.0)..addAll(trackDividers)..add(1.0);
+    double trackSegmentLeft = trackRect.left + allTrackDividers[0] * trackRect.width;
+    double trackSegmentRight;
+    for (int i = 1; i < allTrackDividers.length; i++) {
+      trackSegmentRight = trackRect.left + allTrackDividers[i] * trackRect.width;
+      final double leftHorizontalAdjustment = i != 1 ? horizontalAdjustment : 0.0;
+      final double rightHorizontalAdjustment = i < allTrackDividers.length ? horizontalAdjustment : 0.0;
+      final Rect trackSegmentRect = Rect.fromLTRB(trackSegmentLeft + leftHorizontalAdjustment, trackRect.top, trackSegmentRight - rightHorizontalAdjustment, trackRect.bottom);
+      final Paint paint = trackSegmentsActive[i - 1] ? activePaint : inactivePaint;
+      context.canvas.drawRect(trackSegmentRect, paint);
+      trackSegmentLeft = trackSegmentRight;
+    }
+  }
+}
+
+/// This is the default shape to a [Slider]'s ticks if no other shape is
+/// specified. It paints a solid circle, centered in the on the track.
+/// The color is determined by the [Slider]'s enabled state and the active state
+/// of the track in which it is on. These colors are defined in the
+/// [SliderThemeData].
+///
+/// See also:
+///
+///  * [Slider] for the component that this is meant to display this shape.
+///  * [SliderThemeData] where an instance of this class is set to inform the
```

Comment:
[SliderTheme], which can be used to configure the the tick shape of all sliders in a widget subtree.